js/main.js

```diff
@@ -135,5 +134,39 @@ ul.addEventListener('click', function (event) {
       previewPhoto.setAttribute('src', prePopulate.$url);
     }
   }
+  deleteButton();
   document.querySelector('#formTitle').innerHTML = 'Edit Entry';
 });
+
+const del = document.querySelector('.delete');
+function deleteButton() {
+  del.classList.remove('hidden');
+}
+
+const mod = document.querySelector('.modal');
+del.addEventListener('click', function () {
+  mod.classList.remove('hidden');
+});
+
+const cancel = document.querySelector('.cancel');
+cancel.addEventListener('click', function () {
+  mod.classList.add('hidden');
+});
+
+const confirm = document.querySelector('.confirm');
+confirm.addEventListener('click', function (event) {
```

Comment:
it doesn't seem like the callback function here is actually deleting the `li` from your `data.entries` object in localStorage. 

Try doing this:

1) Use the `filter` method on `data.entries` array, the callback function provided in the `filter` method should check to make sure to that the `entryId` of the index `!==` the `data.editing.entryId`. Once you have the filtered array, assign that value to `data.entries`. By doing this, you are filtering out the index that you are trying to delete from the `data.entries` array, and overwriting `data.entries` with the newly filtered array.

Your code should look something like this: 
```
const filteredArray = data.entries.filter(CALLBACK_FUNCTION_HERE);
data.entries = filteredArray;
```
see this [link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to read up on the filter method.

2) Use a `querySelectorAll` to grab an array of all `li`s on the page. Iterate through that array of `li`s and call the `remove` method on the one that matches the `entryId` that you are trying to delete. It would look like `liElements[i].remove()` inside your `for` loop (assuming that `liElements` is the name of the variable assigned to `document.querySelectorAll('li');`